### DISCUSSION TITLE
Declaration of Spatie jsonSerialize() must be compatible with Illuminate\Support\Collection::jsonSerialize(): array

### DISCUSSION THREAD

**Discussion by gundogduyakici** [spatie/laravel-medialibrary#2967] [Q&A]
**Symfony\Component\ErrorHandler\Error\FatalError**

Declaration of Spatie\MediaLibrary\MediaCollections\Models\Collections\MediaCollection::jsonSerialize() must be compatible with Illuminate\Support\Collection::jsonSerialize(): array

C:\laragon\www\xxxx\vendor\spatie\laravel-medialibrary\src\MediaCollections\Models\Collections\MediaCollection .php : 51

Why am I getting this error?

My php version is 8.0
I use Laravel 9

**Comment by anhzf** [comment]
I'm facing this issue previously, then I upgrade the Laravel version and using the latest version laravel-medialibrary

```bash
composer require laravel/framework:^9.4
composer require spatie/laravel-medialibrary:^10.4.0
```

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-medialibrary/blob/f64d461bff23c9cc097724b54a8f31256f878bb1/src/MediaCollections/Models/Collections/MediaCollection.php#L57
It has the same return type, 
maybe you did have problems with instalation or with cache, try deletin spatie folder and `composer update -W`

**Reply by rezalabbafi** [reply]
I've installed vesrion "10.0.0" 
I've just run composer update -W command 
and this is the class it doesn't return array yet!!!! 
public function jsonSerialize() 
.
.
.
->keyBy('uuid');

**Reply by parallels999** [reply]
@rezalabbafi try to find what do you have on ` old($this->formFieldName ?? $this->collectionName)`
https://github.com/spatie/laravel-medialibrary/blob/ff4a546bafc0ecb4c779055fa1e5850c3b8b2bfe/src/MediaCollections/Models/Collections/MediaCollection.php#L63

**Reply by madfortech** [reply]
composer update -W
 w means ?
